src/hooks/useAuth.ts

```diff
@@ -0,0 +1,108 @@
+import { useContext, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { AuthContext } from "../context";
+import { LogInUserInput, UserDetailType } from "../types";
+import { signupService, logoutService, loginService } from "../services";
+import { auth } from "../firebase-config";
+import { createUserDocument } from "../utils";
+import { getErrorMessage } from "../utils";
+import { onAuthStateChanged } from "firebase/auth";
+
+const useAuth = () => {
+  const { authState, authDispatch } = useContext(AuthContext);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    onAuthStateChanged(auth, currentUser => {
+      (async () => {
+        if (currentUser) {
+          authDispatch({
+            type: "SET_AUTHENTICATION",
+            payload: {
+              isAuthenticated: true,
+              uid: currentUser.uid,
+            },
+          });
+        } else {
+          authDispatch({
+            type: "SET_AUTHENTICATION",
+            payload: {
+              isAuthenticated: false,
+              uid: "",
+            },
+          });
+        }
+      })();
+    });
+  }, [authDispatch]);
```

Comment:
why are you using authDispatch function as dependency?

Reply:
It was giving me waring that's why. I am looking into how to fix the issue. Thanks for the review.